apps/web/src/components/SkillBrowser/SkillBrowser.tsx

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { FieldError, RegisterOptions, useFormContext } from "react-hook-form";
+import get from "lodash/get";
+
+import { getLocalizedName, uiMessages } from "@gc-digital-talent/i18n";
+import { Combobox, Field, Select } from "@gc-digital-talent/forms";
+
+import { BaseSkillBrowserProps } from "./types";
+import {
+  INPUT_NAME,
+  getCategoryOptions,
+  getFamilyOptions,
+  getFilteredFamilies,
+  getFilteredSkills,
+  getSkillFamilySkillCount,
+} from "./utils";
+import NullFamilyMessage from "./NullFamilyMessage";
+
+type SkillBrowserProps = BaseSkillBrowserProps & {
+  name: string;
+  isMulti?: boolean;
+  rules?: RegisterOptions;
+};
+
+const SkillBrowser = ({
+  skills,
+  showCategory,
+  name,
+  rules = {},
+  isMulti = true,
+}: SkillBrowserProps) => {
+  const intl = useIntl();
+  const id = React.useId();
+  const {
+    watch,
+    resetField,
+    register,
+    formState: { errors },
+  } = useFormContext();
+  const skillError = get(errors, name)?.message as FieldError;
+  const inputNames = {
+    category: `${id}-${INPUT_NAME.CATEGORY}`,
+    family: `${id}-${INPUT_NAME.FAMILY}`,
+  };
+  const [category, family] = watch([inputNames.category, inputNames.family]);
+
+  const filteredFamilies = React.useMemo(() => {
+    return getFilteredFamilies({ skills, category });
+  }, [skills, category]);
+
+  const filteredSkills = React.useMemo(() => {
+    return getFilteredSkills({ skills, family, category });
+  }, [category, family, skills]);
+
+  React.useEffect(() => {
+    resetField("skill");
+  }, [category, family, resetField]);
+
+  React.useEffect(() => {
+    resetField(inputNames.family);
+  }, [category, inputNames.family, resetField]);
+
+  const categoryOptions = getCategoryOptions(skills, intl);
+  const familyOptions = getFamilyOptions(skills, intl, category);
+
+  return (
+    <>
+      <div
+        data-h2-display="base(grid)"
+        data-h2-gap="base(x1)"
+        data-h2-margin-bottom="base(x1)"
+        {...(showCategory
+          ? {
+              "data-h2-grid-template-columns": "base(1fr) p-tablet(1fr 1fr)",
+            }
+          : {
+              "data-h2-grid-template-columns": "base(1fr)",
+            })}
+      >
+        {showCategory && (
+          <Select
+            id={inputNames.category}
```

Comment:
What's the reason for the dynamic IDs?  Is this going to make it tricky to do anything with the family or category in the save handler?
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/8f836161-ee22-4a3b-9278-5ccaecf11efc)

Reply:
Oh, this is so we can use multiple on the same page. Yeah, it likely would have an impact if we actually wanted to reference these in the form submission. Though, I believe that we have no plans on doing this and these are just filters for the skill combobox.

@Jerryescandon do you know if we ever need to submit the category or family with this input when it is used?